### PR TITLE
1627-consolidate-heat-agent-code: Consolidate code in HeatEmissionAge…

### PR DESCRIPTION
**Pull request by nagarajankarthik**
…ntGeoJson and HeatAgent into HeatAgent.

### CODE REVIEW

File: Agents/HeatEmissionAgent/Dockerfile [hunk 1]
```diff
@@ -21,11 +21,11 @@ RUN --mount=type=cache,target=/root/.m2/repository mvn package
 
 # Second stage: copy the downloaded dependency into a new image and build into an app
 #==================================================================================================
-FROM tomcat:8.5 as agent
+FROM tomcat:9.0 as agent
 
 WORKDIR /app
 
-COPY --from=builder /root/HeatEmissionAgent/output/heatemissionagent##1.0.0.war $CATALINA_HOME/webapps/
+COPY --from=builder /root/HeatEmissionAgent/output/heat-agent##2.0.0.war $CATALINA_HOME/webapps/
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/Dockerfile:28]: Be consistent with the naming of the agent. Either "HeatEmissions Agent" or "Heat Agent". The former is more descriptive in my opinion

File: Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml [hunk 2]
```diff
@@ -0,0 +1,192 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>heat-agent</artifactId>
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml:7]: Same as previous comment on naming

File: Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    <packaging>war</packaging>
+    
+    <!-- Project properties -->
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+        
+        <!-- Version of the JPS Base Library to use -->
+        <jps.base.version>1.35.0</jps.base.version>
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml:18]: Use the latest version of base lib

File: Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>1.16.2</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+		  <groupId>com.jayway.jsonpath</groupId>
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml:185]: Minor issue but could you fix the indentation?

File: Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatEmissionAgent.java [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+		FuelCEIEffiQuery.append("?cei om:hasNumericalValue ?CEI .");
+		FuelCEIEffiQuery.append("?cei om:hasUnit ?a .");
+		FuelCEIEffiQuery.append("?a om:symbol ?unit .}");
+		JSONArray plantInfoQueryResult = AccessAgentCaller.queryStore("jibusinessunits", FuelCEIEffiQuery.toString());
+		return plantInfoQueryResult;
+	}
+
+	// Geometric coordination query
+	public static JSONArray CoordinateQuery(String CityFurnitureIRI) {
+		// StringBuffer coordinateQuery = new StringBuffer("PREFIX ocgml:
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatEmissionAgent.java:250]: remove any unused code/commented out code

File: Agents/HeatEmissionAgent/README.md [hunk 6]
```diff
@@ -33,20 +33,19 @@ To build and start the agent, open up the command prompt in the same directory a
...
+
+curl -X POST -H "Content-Type: application/json" -d '{"job":{"lower_bounds":"8464#23588#0","upper_bounds":"17619#30520#105"}}' http://localhost:8084/heat-agent/performheatquery
 
-POST http://localhost:1080/heatemissionagent/performheatquery
-Content-Type: json
-{"job":{"lower_bounds":"8464#23588#0","upper_bounds":"17619#30520#105"}}
 
 If the agent runs successfully, you should see a returned JSON Object that is similar to the one shown below.
 
 {"result":[{"Coordinate":"13469.086796413478#26462.33106185692#41.0","Heat Emission":80.22970472647032},{"Coordinate":"13252.157092925416#28023.664545558153#25.0","Heat Emission":80.22970472647032},{"Coordinate":"13341.125256486484#26898.004451070658#21.0","Heat Emission":80.22970472647032}]}
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/README.md:47]: Can the results and error be better represented using markdown, perhaps using inline code?

File: Agents/HeatEmissionAgent/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+version: "3.8"
+
+services:
+  heat-agent:
+    image: heat-agent:2.0.0
```
Comment (sganguly-cares) [inline Agents/HeatEmissionAgent/docker-compose.yml:5]: standardise the agent name as mentioned before

**Review by sganguly-cares — CHANGES_REQUESTED** [review]
General comment - The agent name is not consistent across the folders, pom.xml, docker image etc. This should be fixed.

File: Agents/HeatEmissionAgent/Dockerfile (hunk 1 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/Dockerfile]: The 'heatemissionagent' label is now used in all files.

File: Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml (hunk 2 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml]: Done

File: Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml (hunk 3 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml]: I updated it to 1.39.0

File: Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml (hunk 4 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/HeatEmissionAgent/pom.xml]: Done

File: Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatEmissionAgent.java (hunk 5 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatEmissionAgent.java]: Done

File: Agents/HeatEmissionAgent/docker-compose.yml (hunk 7 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/docker-compose.yml]: Done

File: Agents/HeatEmissionAgent/README.md (hunk 6 above)
Comment (nagarajankarthik) [reply Agents/HeatEmissionAgent/README.md]: I added markdown to represent the results and error in a separate block of text.

**Review by sganguly-cares — APPROVED** [review]

**Pull request merged**